pyxdf/pyxdf.py

```diff
@@ -62,6 +62,7 @@ def __init__(self, xml):
 
 
 def load_xdf(filename,
+             load_only=None,
```

Comment:
I would prefer an argument named 'extract_streams', and for the preferred argument value to be a list of dicts, where each dict has to have at least one of the keys "name", "type", "source_id".
This mirrors stream resolving in liblsl.

Reply:
Sure, we can discuss the parameter name; `load_only` is not really a good choice, but I don't like `extract_streams` either. This doesn't work nicely with `None`, because `extract_streams=None` sounds like we don't want to extract any stream. With `load_only=None`, at least to me it sounds like we don't want to load any specific stream, so we load all. What about `load_streams` instead?

Reply:
Also, for the same reason as before (namely that the only unique value associated with a stream is its `stream_id`), I'd very much prefer not to have the option to provide other fields. This makes parsing and error-handling a lot more complicated, and is also a lot more complicated from a user's perspective.

Reply:
A compromise could be a helper function to convert the liblsl-like keywords into a streamid for a given file.

Reply:
> A compromise could be a helper function to convert the liblsl-like keywords into a streamid for a given file.

Sure, this would be an option.

> I agree with @cboulay and would furthermore say that a liblsl-like approach feels to me actually less complicated from an end-users perspective. Similar to #23, streamids are arbitrary and can change between sessions. If the streams can not be uniquely resolved by keywords, i would expect to load all matching.

Only because you are using LSL. I only use XDF files without LSL, so it is much more confusing to me. Furthermore, stream IDs are arbitrary and can change between sessions is not a good argument here, because we only deal with a single XDF file here - stream IDs do not change within a single file.

Reply:
> stream IDs do not change within a single file.
But between files, so if i analyze multiple files over the course of weeks (which we often do), i would have to catch any changing streamids, while giving a dictionary of keywords is native and not arbitrary. Anyways, this is a functionality our lab will use for sure, so i wrote a quick parser/matcher for that:
It does retrieve a) the stream info from a file and b) return streamids based on matching a list of dictionary with those. Tests are in /pyxdf/test/test_resolve_stream_id. Please be aware the the filepath to the minimal test.xdf was hardcoded - sorry for the hack.
https://github.com/xdf-modules/xdf-Python/tree/resolve_streamid

Looking forward to your feedback.

Reply:
Sure, I'm not saying that this is not important, it just should be handled outside `load_xdf`, which loads a single XDF file.

I already have something along these lines here: https://github.com/cbrnr/mnelab/blob/master/mnelab/utils/xdf.py
Basically, you would `parse_chunks(parse_xdf(xdffile))` to get a list of important fields for all streams. From this list, you could resolve on whatever field you want.

Reply:
How about `stream_ids` as a keyword? I think this will help get rid of some confusion among the vast majority of xdf users who are coming from LSL.

Reply:
+1 for `stream_ids`!